pkgs/applications/science/math/scilab-bin/default.nix

```
@@ -1,31 +1,50 @@
-{ stdenv, fetchurl, lib, xorg }:
+{ lib, stdenv, fetchurl, undmg, makeWrapper, xorg }:
 
 let
-  badArch = throw "scilab-bin requires i686-linux or x86_64-linux";
-
-  architecture =
-    if stdenv.hostPlatform.system == "i686-linux" then
-      "i686"
-    else if stdenv.hostPlatform.system == "x86_64-linux" then
-      "x86_64"
-    else
-      badArch;
-in
-stdenv.mkDerivation rec {
   pname = "scilab-bin";
   version = "6.1.1";
 
-  src = fetchurl {
-    url = "https://www.scilab.org/download/${version}/scilab-${version}.bin.linux-${architecture}.tar.gz";
-    sha256 =
-      if stdenv.hostPlatform.system == "i686-linux" then
-        "0fgjc2ak3b2qi6yin3fy50qwk2bcj0zbz1h4lyyic9n1n1qcliib"
-      else if stdenv.hostPlatform.system == "x86_64-linux" then
-        "sha256-PuGnz2YdAhriavwnuf5Qyy0cnCeRHlWC6dQzfr7bLHk="
-      else
-        badArch;
+  srcs = {
+    aarch64-darwin = fetchurl {
+      url = "https://www.utc.fr/~mottelet/scilab/download/${version}/scilab-${version}-accelerate-arm64.dmg";
+      sha256 = "sha256-L4dxD8R8bY5nd+4oDs5Yk0LlNsFykLnAM+oN/O87SRI=";
+    };
+    x86_64-darwin = fetchurl {
+      url = "https://www.utc.fr/~mottelet/scilab/download/${version}/scilab-${version}-x86_64.dmg";
+      sha256 = "sha256-tBeqzllMuogrGcJxGqEl2DdNXaiwok3yhzWSdlWY5Fc=";
+    };
+    x86_64-linux = fetchurl {
+      url = "https://www.scilab.org/download/${version}/scilab-${version}.bin.linux-x86_64.tar.gz";
+      sha256 = "sha256-PuGnz2YdAhriavwnuf5Qyy0cnCeRHlWC6dQzfr7bLHk=";
+    };
+  };
+  src = srcs.${stdenv.hostPlatform.system};
```

```suggestion
  src = srcs.${stdenv.hostPlatform.system} or (throw "unsupported system ${stdenv.hostPlatform.system}");
```
otherwise this will generate syntax errors if evaluated on other platforms.